### PR TITLE
Big documentation update and adjustment to html method

### PR DESCRIPTION
**Pull request by amafehr**
# summary
This PR cleans up experimental files, updates documentation, and lets users save only html (instead of automatically making a pdf).

## Description
- Adds `raw_requirements.txt` as a precursor to formally packaging this (we still need to separately figure this out for a mixed ecosystem of python and javascript).
- removes `proof_of_concept/` and experimental pandoc and weasyprint results (but keeps notes on these incase others want to chase down these routes). 
- massively updates documentation, to include better install instructions and updated decisions. 
- adds an arg to the `generate_svg.py` allowing users to only save `.html` instead of forcing a `.pdf`. Although I don't like the growing list of args, this lets users produce the plots' code in a much more performant way, and they can optionally choose their own rendering method outside of ours (if it's problematic). 
- adds a few test files that @jstonge is working with to have contour lines working. We will add some documentation and a sample.pdf here with contour lines once we figure out how to integrate into py-allotax.

## Notes
- I am increasingly not convinced that `parser = argparse.ArgumentParser(...)` is needed here (overkill?)

## Testing
- If you already have the repo, `git fetch --all` to get branch changes. 
- `git checkout image-gen-method`
- May need to install all `npm` requirements again
- `cd src`
- `python3 generate_svg.py convert/boys_2022.json convert/boys_2023.json output_charts/test.pdf "0.17" --desired_format "html"`
- `python3 generate_svg.py convert/boys_2022.json convert/boys_2023.json output_charts/test.pdf "0.17"`
- *UPDATE*: you can just run `examples.ipynb`

### CODE REVIEW

File: DECISIONS.md [hunk 1]
```diff
@@ -4,44 +4,51 @@
...
 
+ ### HTML to image options
+
+ This has to be lightweight; many options open a headless browser reliant on a browser driver (e.g. selenium uses the Chrome driver) which takes a few seconds per conversion (see posted issue for follow-up on this separately). We explored numerous options that do not seem possible, so this area remains a task anyone is free to take on. Specifically, we tried `pandoc`, `weasyprint`, node.js `html-to-pdf`, and a less restrictive DOM `linkedom`.
+ - `pandoc`: would have been a good option but had many texlive (latex universe) and system dependencies that ultimately did not work out.
+ - `weasyprint`: we think it could work but may require some tinkering with the CSS elements in the HTML plot template. Library logging showed weasyprint fails to render SVG image and an option wasn't apparent.
+ - `linkedom`: As with the current DOM (`jsdom`), a richer browser was required (headless browser to render styles and flexboxes), making `html-to-pdf` a non-option.
 
 ## Future improvements (TODOs):
```
Comment (jstonge) [inline DECISIONS.md:48]: We might just want to point to a file, github project, or issue. If no, I fear that this list will be outdated faster than you can say... Jack Robinson ? Is that a real idiom.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,60 @@
...
+1. One package depends on having Chrome (brower) or Chromium (browser driver) installed. **If you have Chrome installed, you can skip this step**. Otherwise, install Chrome or Chromium.
 Here is the link for the [Chromium download](https://www.chromium.org/getting-involved/download-chromium/).
 
-1. Install the following:
+1. Clone this repo and install the requirements:
     ```shell
     git clone https://github.com/carterwward/py-allotax.git &&
     cd py-allotax &&
-    npm install d3 jsdom jstonge/allotaxonometer#py-allotax-experimental &&
+    npm install @ungap/structured-clone d3 jsdom jstonge/allotaxonometer#py-allotax-experimental &&
```
Comment (jstonge) [inline README.md:36]: I need to check, but if I merge `py-allotax-experimental` branch to the main, then publish it to npm. We could simply do `npm install allotaxonometer jsdom` (ignoring `d3` and `structured-clone`, as they are deps in `allotaxonometer`.

File: README.md [hunk 3]
```diff
@@ -1,39 +1,60 @@
...
     pip install pandas pyhtml2pdf
     ```
-(*note: we are in the process of replacing the HTML->PDF generation method to remove `pyhtml2pdf` dependency*)
+(*note: we will eventually package this repo and streamline installation. See `raw_requirements.txt` for versions and notes if trouble abounds.*)
 
 ## Usage instructions
-1. Verify your data is in the required format by seeing the example in `data/test_data`.
+1. Verify your data is in the required format (`.json`) by seeing examples in `convert/test_data`.
```
Comment (jstonge) [inline README.md:42]: we should add test for that, assuming users will fail to do it right.

File: proof_of_concept/DoughnutChart.js
Comment (jstonge) [inline proof_of_concept/DoughnutChart.js:1]: good ridance

File: README.md [hunk 4]
```diff
@@ -1,39 +1,60 @@
-# Allotax through Python: proof of concept
+# Allotaxonometer through Python
+
+The `py-allotax` implements a python interface to the `allotaxonometer` library. The tool here provides a way for users to input data and arguments and receive back a saved plot! The tool is designed to be used in a command line or in a python notebook in a few lines of code (see usage instructions at the bottom).
+
+## Structure
```
Comment (jstonge) [inline README.md:5]: I would put `installation steps` section first.

File: README.md
Comment (jstonge) [inline README.md:1]: Maybe we could add an `examples/` directory to start writing simple case studies of how to use the package?

Comment (amafehr) [reply README.md:1]: I had some trouble with an examples directory--I could work out the python relative imports, but JS modules were being funky. I think it'll resolve when we package this. For now, see `examples.ipynb` in `src`. Everything there should run.

File: README.md (hunk 3 above)
Comment (amafehr) [reply README.md:41]: I added a function in `utils`

File: README.md (hunk 2 above)
Comment (amafehr) [reply README.md:22]: Sounds good, happy to verify as a user on my end when we get to that point!

**Pull request merged**